### DISCUSSION TITLE
Problems after Sonos app update

### DISCUSSION THREAD

**Discussion by SenorRossi71** [hklages/node-red-contrib-sonos-plus#298] [General]
Hello, 

yesterday I started the app update. Since then I can no longer operate my speakers with node-red. (System is: raspberrymatic with redmatic)
The error message is: 

processInputMsg:(0 , crypto_1.randomUUID) is not a function :: Details: {"stack":"TypeError: (0 , crypto_1.randomUUID) is not a function\n    at new SonosDeviceBase (/usr/local/addons/redmatic/var/node_modules/@svrooij/sonos/lib/sonos-device-base.js:37:67)\n    at new SonosDevice (/usr/local/addons/redmatic/var/node_modules/@svrooij/sonos/lib/sonos-device.js:37:9)\n    at processInputMsg (/usr/local/addons/redmatic/var/node_modules/node-red-contrib-sonos-plus/src/sonos-manage-mysonos.js:187:22)\n    at SonosManageMySonosNode._inputCallback (/usr/local/addons/redmatic/var/node_modules/node-red-contrib-sonos-plus/src/sonos-manage-mysonos.js:124:11)\n    at /usr/local/addons/redmatic/lib/node_modules/node-red/node_modules/@node-red/runtime/lib/nodes/Node.js:210:26\n    at Object.trigger (/usr/local/addons/redmatic/lib/node_modules/node-red/node_modules/@node-red/util/lib/hooks.js:166:13)\n    at SonosManageMySonosNode.Node._emitInput (/usr/local/addons/redmatic/lib/node_modules/node-red/n...

any Idea to solve this error?
Thanks!

**Comment by hklages** [comment]
hi. Can you please check every player and ensure that every one has the
newest update. Today I also had a problem because an update was available
and not applied.

**Reply by SenorRossi71** [reply]
Hey, 
all Players has Version 16.2 (build 79.0-52294)
the app told me that there are no updates necessary

**Comment by hklages** [comment]
Ok - thats seems to be the right version. 

What does it mean: Raspberrymatic with redmatic. Usually redmatic runs on a CCU as add on. Raspberrymatic usually does not run on a CCU.  

Redmatic is difficult. What is your Node.js and npm version?

**Reply by SenorRossi71** [reply]
RaspberryMatic is OpenSource based homematicIP CCU, it runs on a raspberryPi
Redmatic is an CCU adon to run nodered on a CCU

Versions are:
Node.js 20.9.0
npm 10.1.0

**Comment by hklages** [comment]
Understand. Node.js and npm versions are good.

Please check a simple flow with "inject -> sonos-plus and command "household.get.groups"  -> debug "and a config node with a player where you have verified the ip address with `http://<your-ip-address>:1400/info`.  
Same error message?

**Reply by SenorRossi71** [reply]
Here the test Node after the frist deploy
![First](https://github.com/hklages/node-red-contrib-sonos-plus/assets/60069458/c5cf1bcc-9629-4c3f-8cf0-e23483e9e220)

and her after clicking the inject:
![after](https://github.com/hklages/node-red-contrib-sonos-plus/assets/60069458/9cbe706f-f886-4498-a108-8d0e6d1eb837)

same as bevor

**Comment by hklages** [comment]
What means "started the app update"? Did you install the new SONOS Android app. That should not be a a problem - I did the same two days ago.

**Reply by SenorRossi71** [reply]
I made an iPhone- App update 
I think the sonso system itself was updated too then?
Version shown in the app:
Sonos System S2
buildt 80.00.08

**Reply by hklages** [reply]
Yeah - the SONOS Player also got an update. But that should not cause the problems with Node-RED and the sonos-plus package. I assume there is something else what changed. Any network configuration changes - router? DNS server?

**Comment by hklages** [comment]
Are you able to use the shell? 

```
$ source /usr/local/addons/redmatic/home/.profile
$ cd /usr/local/addons/redmatic/var
$ npm --version
$ npm list
```

**Reply by SenorRossi71** [reply]
Now I have done a complete RedMatic update. Many for this also to [https://homematic-forum.de/forum/viewtopic.php?t=81303#p792296](url) !!!
Many thanks also to @hklages !!! The SONOS package is realy TOP!!!

**Comment by hklages** [comment]
Thanks for the feedback. Good to hear that everything works.
